crates/ruff/src/linter.rs

```diff
@@ -353,7 +354,7 @@ pub fn lint_only(
         &directives,
         settings,
         noqa,
-        None,
+        source_kind,
```

Comment:
Should this _not_ be optional? Seems like maybe it should be required everywhere 🤔

Reply:
Yes, it should be. It probably needs to be defined at a top level and I've a local branch where I worked on that a bit. I'll be looking into it soon.